pkg/agent/proxy/endpoints.go

```diff
@@ -95,10 +108,42 @@ func (t *endpointsChangesTracker) OnEndpointUpdate(previous, current *corev1.End
 	return len(t.changes) > 0
 }
 
+// EndpointSliceUpdate updates the given service's endpoints change map based on the <previous, current> endpoints pair.
+// It returns true if items changed, otherwise it returns false. Will add/update/delete items of endpointsChange Map.
+// If removeSlice is true, slice will be removed, otherwise it will be added or updated.
+func (t *endpointsChangesTracker) OnEndpointSliceUpdate(endpointSlice *discovery.EndpointSlice, removeSlice bool) bool {
+	// This should never happen.
+	if endpointSlice == nil {
+		klog.Error("Nil endpointSlice passed to EndpointSliceUpdate")
+		return false
+	}
+
+	if _, has := supportedEndpointSliceAddressTypes[endpointSlice.AddressType]; !has {
+		klog.V(4).Infof("EndpointSlice address type not supported: %s", endpointSlice.AddressType)
+		return false
+	}
+
+	if _, _, err := endpointSliceCacheKeys(endpointSlice); err != nil {
+		klog.Warningf("Error getting endpoint slice cache keys: %v", err)
```

Comment:
Better to use the same format for EndpointService (rather than all endpointSlice, EndpointSlice, endpoint slice..)